datastore/storagebackend/postgresql/getextents.go

```diff
@@ -0,0 +1,71 @@
+package postgresql
+
+import (
+	"database/sql"
+	"datastore/datastore"
+	"fmt"
+	"time"
+
+	_ "github.com/lib/pq"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// getTimeExtent gets the current time extent of all observations in the storage.
+func getTimeExtent(db *sql.DB) (*datastore.TimeInterval, error) {
+	query := "SELECT min(obstime_instant), max(obstime_instant) FROM observation"
+	row := db.QueryRow(query)
+
+	var start, end time.Time
+
+	err := row.Scan(&start, &end)
+	if err != nil {
+		return nil, fmt.Errorf("row.Scan() failed: %v", err)
+	}
+
+	return &datastore.TimeInterval{
+		Start: timestamppb.New(start),
+		End:   timestamppb.New(end),
+	}, nil
+}
+
+// getGeoExtent gets the current geo extent of all observations in the storage.
+func getGeoExtent(db *sql.DB) (*datastore.BoundingBox, error) {
+	query := `
+		SELECT ST_XMin(ext), ST_YMin(ext), ST_XMax(ext), ST_YMax(ext)
+		FROM (SELECT ST_Extent(point::geometry) AS ext FROM geo_point) t
```

Comment:
Nice way to do this!

Reply:
🙂 Not sure if this is the best idiom (wrt. performance etc.) in this case, but it looks nice, I agree.